src/java/com/rapleaf/jack/queries/IndexHints.java

```diff
@@ -5,52 +5,52 @@
   private IndexHints() {
   }
 
-  public static IndexHint use(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.ALL, index, indices);
+  public static IndexHint use(Index... indices) {
+    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.ALL, indices);
   }
 
-  public static IndexHint useForJoin(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.JOIN, index, indices);
+  public static IndexHint useForJoin(Index... indices) {
+    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.JOIN, indices);
   }
 
-  public static IndexHint useForOrderBy(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.ORDER_BY, index, indices);
+  public static IndexHint useForOrderBy(Index... indices) {
+    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.ORDER_BY, indices);
   }
 
-  public static IndexHint useForGroupBy(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.GROUP_BY, index, indices);
+  public static IndexHint useForGroupBy(Index... indices) {
+    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.GROUP_BY, indices);
   }
 
-  public static IndexHint force(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.ALL, index, indices);
+  public static IndexHint force(Index... indices) {
+    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.ALL, indices);
   }
 
-  public static IndexHint forceForJoin(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.JOIN, index, indices);
+  public static IndexHint forceForJoin(Index... indices) {
+    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.JOIN, indices);
   }
 
-  public static IndexHint forceForOrderBy(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.ORDER_BY, index, indices);
+  public static IndexHint forceForOrderBy(Index... indices) {
+    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.ORDER_BY, indices);
   }
 
-  public static IndexHint forceForGroupBy(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.GROUP_BY, index, indices);
+  public static IndexHint forceForGroupBy(Index... indices) {
```

Comment:
My main issue with this pattern is that it adds a lot of unnecessary complication to the code. It forces us to add extra overloads and extra lines to deal with the single arg and the varargs separately. Look for example at the constructor for WhereOperator, where we have three overloads all doing the same thing. There are also cases like in In/NotIn where we alternatively accept a Collection, and there is no way we can validate that at compile time. We could add runtime checks if you really want this validated to have at least one argument.